### ISSUE TITLE
[Feature Request] ES6 Build

### ISSUE THREAD

**Issue by NicolasParada** [developit/preact-router#91]
I noticed that `preact-router` exposes jsnext:main but it isn't quite ES6. There you have JSX syntax and object rest destructuring in function params, example:

```jsx
const Link = ({ children, ...props }) => (
  <a {...props} onClick={handleLinkClick}>{ children }</a>
);
```

**Comment by NicolasParada** [comment]
Because it isn't much code, it could get rid of that syntax completely

```js
const Link = props => {
  const rest = Object.assign({}, props, { onClick: handleLinkClick });
  delete rest.children;
  return h('a', rest, props.children);
}
```

```js
const Route = ({ component: RoutedComponent, url, matches }) => {
  return h(RoutedComponent, { url, matches });
}
```

**Comment by developit** [comment]
I think it's still worth avoiding relying on `Object.assign` directly for browser support. Maybe we have Rollup output ES Modules too?

**Comment by NicolasParada** [comment]
It's an ES6 feature. I think it's ok for the jsnext:main.

**Comment by developit** [comment]
True, and it can be transpiled.. so no jsx?

**Comment by NicolasParada** [comment]
no jsx

**Comment by developit** [comment]
K. Wasn't much of it in there anyway.

**Comment by NicolasParada** [comment]
Oh, right. Babel is not gonna transpile the Object.assign (for the main build)...

**Comment by developit** [comment]
I think there's a way to, or maybe switch to Bublé...

**Comment by developit** [comment]
Fixed via #93!
